src/EditorFeatures/Core/Wrapping/BinaryExpression/BinaryExpressionCodeActionComputer.cs

```diff
@@ -0,0 +1,119 @@
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Wrapping.BinaryExpression
+{
+    internal partial class AbstractBinaryExpressionWrapper<TBinaryExpressionSyntax> where TBinaryExpressionSyntax : SyntaxNode
+    {
+        private class BinaryExpressionCodeActionComputer : 
+            AbstractCodeActionComputer<AbstractBinaryExpressionWrapper<TBinaryExpressionSyntax>>
+        {
+            private readonly ImmutableArray<SyntaxNodeOrToken> _exprsAndOperators;
+            private readonly SyntaxTriviaList _indentationTrivia;
+
+            private readonly SyntaxTriviaList _newlineBeforeOperatorTrivia;
+
+            public BinaryExpressionCodeActionComputer(
+                AbstractBinaryExpressionWrapper<TBinaryExpressionSyntax> service,
+                Document document,
+                SourceText originalSourceText,
+                DocumentOptionSet options,
+                TBinaryExpressionSyntax binaryExpression,
+                ImmutableArray<SyntaxNodeOrToken> exprsAndOperators,
+                CancellationToken cancellationToken)
+                : base(service, document, originalSourceText, options, cancellationToken)
+            {
+                _exprsAndOperators = exprsAndOperators;
+
+                var generator = SyntaxGenerator.GetGenerator(document);
+                var indentationString = OriginalSourceText.GetOffset(binaryExpression.Span.Start)
+                                                          .CreateIndentationString(UseTabs, TabSize);
+
+                _indentationTrivia = new SyntaxTriviaList(generator.Whitespace(indentationString));
+                _newlineBeforeOperatorTrivia = service.GetNewLineBeforeOperatorTrivia(NewLineTrivia);
+            }
+
+            protected override async Task<ImmutableArray<WrappingGroup>> ComputeWrappingGroupsAsync()
+            {
+                var actions = ArrayBuilder<WrapItemsAction>.GetInstance();
+
+                actions.Add(await GetWrapCodeActionAsync(includeOperators: false).ConfigureAwait(false));
+                actions.Add(await GetWrapCodeActionAsync(includeOperators: true).ConfigureAwait(false));
+                actions.Add(await GetUnwrapCodeActionAsync().ConfigureAwait(false));
+
+                return ImmutableArray.Create(new WrappingGroup(
+                    isInlinable: true, actions.ToImmutableAndFree()));
+            }
+
+            private Task<WrapItemsAction> GetWrapCodeActionAsync(bool includeOperators)
+                => TryCreateCodeActionAsync(
+                    GetWrapEdits(includeOperators),
+                    FeaturesResources.Wrapping,
+                    includeOperators
+                        ? FeaturesResources.Wrap_expression_including_operators
+                        : FeaturesResources.Wrap_expression);
+
+            private ImmutableArray<Edit> GetWrapEdits(bool includeOperators)
```

Comment:
💡 Short documentation for `includeOperators` could be helpful

Reply:
done!